services/graph/pkg/identity/backend.go

```diff
@@ -40,6 +41,10 @@ type Backend interface {
 	AddMembersToGroup(ctx context.Context, groupID string, memberID []string) error
 	// RemoveMemberFromGroup removes a single member (by ID) from a group
 	RemoveMemberFromGroup(ctx context.Context, groupID string, memberID string) error
+	// RemoveEntryByDNAndAttributeFromEntry creates a request to remove a single member entry by attribute and DN from an ldap entry
+	RemoveEntryByDNAndAttributeFromEntry(entry *ldap.Entry, dn string, attribute string) (*ldap.ModifyRequest, error)
+	// ExpandLDAPAttributeEntries reads an attribute from an ldap entry and expands to users
+	ExpandLDAPAttributeEntries(ctx context.Context, e *ldap.Entry, attribute string) ([]*ldap.Entry, error)
```

Comment:
I think these two should not be part of the interface. They are LDAP specific and the whole point of the backend interface is to allow other (non-LDAP) backends for the users and groups endpoints.

Are they used anywhere outside the LDAP backend itself? It looks to me as they could even stay private methods.